io_scene_godot/converters/mesh.py

```diff
@@ -279,37 +308,40 @@ def generate_lines(self):
 
     def _get_bone_arrays(self):
         """Returns the most influential bones and their weights"""
-        if self.armature is None:
-            return [
-                Array("null, ; No Bones", "", ""),
-                Array("null, ; No Weights", "", "")
-            ]
-
-        # Skin Weights!
-        float_values = Array("FloatArray(")
-        float_valuesw = Array("FloatArray(")
+        has_bone_array = False
+
+        # Skin Weights
+        bone_array = Array("IntArray(")
+        weights_array = Array("FloatArray(")
         for vert in self.vertices:
-            # skin_weights_total += len(v.weights)
             weights = []
-            for i in len(vert.bones):
-                weights += (vert.bones[i], vert.weights[i])
+            for i in range(len(vert.bones)):
+                weights.append((vert.bones[i], vert.weights[i]))
 
-            weights = sorted(weights, key=lambda x: -x[1])
+            if weights:
+                has_bone_array = True
+
+            weights = sorted(weights, key=lambda x: x[1], reverse=True)
             totalw = 0.0
             for weight in weights:
                 totalw += weight[1]
             if totalw == 0.0:
                 totalw = 0.000000001
 
-            for i in range(4):
+            for i in range(MAX_BONE_PER_VERTEX):
                 if i < len(weights):
-                    float_values.append(weights[i][0])
-                    float_valuesw.append(weights[i][1]/totalw)
+                    bone_array.append(weights[i][0])
+                    weights_array.append(weights[i][1]/totalw)
                 else:
-                    float_values.append(0)
-                    float_valuesw.append(0.0)
-
-        return float_values, float_valuesw
+                    bone_array.append(0)
+                    weights_array.append(0.0)
+
+        if has_bone_array:
+            return bone_array, weights_array
+        return [
```

Comment:
If possible, put the no-bones case before iterating through vertices. On high poly models the time to iterate through all the vertices/bones may be significant.